### ISSUE TITLE
C# bindings not compatible of SWIG 4.0.0

### ISSUE THREAD

**Issue by rouault** [OSGeo/gdal#1726]
Reported initially in #1725
Failures like
```
[00:32:58] 	swig.exe -DSWIG2_CSHARP -c++ -csharp -namespace OSGeo.GDAL  -dllimport gdal_wrap -I../../include/csharp -o gdal_wrap.cpp ../../include/gdal.i
[00:32:59] ..\..\include\csharp\swig_csharp_extensions.i(204) : Error: A deprecated csdestruct typemap was found for RasterIOExtraArg, please remove it and replace all csdestruct, csdestruct_derived and csfinalize typemaps by the csdispose, csdispose_derived, csdisposing and csdisposing_derived typemaps.
```

**Comment by runette** [comment]
@rouault - I was reviewing issues to see if there was anything else that should be included in #5046 .

Since the CMake scripts are building on SWIG4 - should this issues be closed?

**Comment by rouault** [comment]
Let's close this, although it doesn't look we know why this works now...
